static/app/components/charts/chartWidgetLoader.tsx

```diff
@@ -13,6 +13,289 @@ interface Props extends LoadableChartWidgetProps {
   id: string;
 }
 
+// We need to map the widget id to the dynamic import because we want the import paths to be statically analyzable.
+const CHART_MAP: Map<
+  string,
```

Comment:
I think we should derive a string literal union type from the keys. Can’t use maps for that, but switching to an object would work, see:

https://www.typescriptlang.org/play/?#code/FAFwngDgpgBAMgewIYBMkCMA2UDCALJAJxAHUBLFAcyhAAVCEIBnGAXhgFcA7Aay4QDuXYAGMEXJiBgBRAGrSAcgBUA+gHEASgEFaACRUkAkgBE101SbYwA5HMWrNO-UdPmVJ68FHjJMHLq0NVQBZHSsAb2AYaJl5ZXVtPQMTMwtjAC4YAAoASjYAPhgyAFsIBGIs6yYoLhBCMAB6ADcyKAEmBrImJg4oYxokMkwOyUIoJGLMMi4oBqgmmpA1QiQIPHIqGmscgBoomOtuPHHMEDwwAGUobrIffCJSCmoQa0zcgqLS8pAs-ZiD6q1erNVrtToSMiUPAgDpiYrFcQNOFlGa1DoCJ40DpHE5nS7XJi3CT3YgbZ6ef7RXbAAC+MCYSBAXQAZq0WBooGJCCgADz7UbTSh7aLvViFegIYpdKA88IoKDMpAcU6ZDlIEQgAB0ADEcDzEKgMNgSY9NnQGMx8jT8sAbaBILATYYUFYeFAwAhmTBwNBPX4AkEVKFaABuYANBr-AB6AH4vMAgA